res/controllers/common-controller-scripts.js

```diff
@@ -190,6 +192,7 @@ const deepMerge = function(target, source) {
     }
 };
 
+/** @deprecated Use {@link Object.assign} instead */
 script.deepMerge = deepMerge;
```

Comment:
Damn this file need a big refactoring…

Reply:
can you elaborate?

Reply:
Everything is declared using `var` sothat variables endup in the global scope. `script` should be declared as such:

```js
(function(global) {
    const script = Object.freeze({
        LIBRARY_COLUMNS: Object.freeze({ … }),
        debug() { … }
    });
})(this)
```

That would improve readability IMHO.

Maybe even [`globalThis`](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/globalThis) can be used? I need to check if implemented in `QJSEngine`.

Reply:
Also `print()` is stillused everywhere (could be remove in 2.60 too)./

Reply:
I assume you mean `global.script = ...`? But yeah, employing that pattern would probably be better, even though I don't find it to improve readability particularly.

Reply:
> Damn this file need a big refactoring…

Are you aware of #11625? Help to fix the build failure there is appreciated.